diesel/src/pg/types/numeric.rs

```diff
@@ -50,14 +50,20 @@ mod bigdecimal {
             digits.reverse();
             let digits_after_decimal = scale as u16 / 4 + 1;
             let weight = digits.len() as i16 - digits_after_decimal as i16 - 1;
-            let index_of_decimal = (weight + 1) as usize;
 
-            let unneccessary_zeroes = digits[index_of_decimal..]
+            let unnecessary_zeroes = if weight >= 0 {
+                let index_of_decimal = (weight + 1) as usize;
+                digits[index_of_decimal..]
```

Comment:
How about we use `.get` here and ~~fail gracefully or~~ panic with a custom message like we do above?